GetMessages/handler.ts

```diff
@@ -50,42 +67,110 @@ type IGetMessagesHandlerResponse =
  * GetMessages expects a FiscalCode as input and returns the Messages
  * as output or a Validation error.
  *
- * TODO: add full results and paging
  */
 type IGetMessagesHandler = (
-  fiscalCode: FiscalCode
+  fiscalCode: FiscalCode,
+  maybePageSize: O.Option<NonNegativeInteger>,
+  maybeEnrichResultData: O.Option<boolean>,
+  maybeMaximumId: O.Option<NonEmptyString>,
+  maybeMinimumId: O.Option<NonEmptyString>
 ) => Promise<IGetMessagesHandlerResponse>;
 
 /**
  * Handles requests for getting all message for a recipient.
  */
-export function GetMessagesHandler(
-  messageModel: MessageModel
-): IGetMessagesHandler {
-  return async fiscalCode => {
-    return pipe(
-      messageModel.findMessages(fiscalCode),
-      TE.map(flattenAsyncIterator),
-      TE.map(_ => filterAsyncIterator(_, isRight)),
-      TE.map(_ => mapAsyncIterator(_, e => e.right)),
-      TE.map(_ => filterAsyncIterator(_, RetrievedNotPendingMessage.is)),
-      TE.map(_ => mapAsyncIterator(_, retrievedMessageToPublic)),
-      TE.bimap(
-        failure => ResponseErrorQuery(failure.kind, failure),
-        ResponseJsonIterator
-      ),
-      TE.toUnion
-    )();
-  };
-}
+export const GetMessagesHandler = (
+  messageModel: MessageModel,
+  serviceModel: ServiceModel,
+  blobService: BlobService
+): IGetMessagesHandler => async (
+  fiscalCode,
+  maybePageSize,
+  maybeEnrichResultData,
+  maybeMaximumId,
+  maybeMinimumId
+) =>
+  pipe(
+    TE.Do,
```

Comment:
Just a hint: you can also use a `sequenceS` to achieve the same result (both solutions are euqual in my opinion)

Reply:
Yes, I used it for a similar case. I think Do is a little bit more explicit in intent.